File: src/main/java/com/eager/questioncloud/point/entity/UserPointEntity.java
package com.eager.questioncloud.point.entity;

import com.eager.questioncloud.point.model.UserPoint;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@Table(name = "user_point")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class UserPointEntity {
    @Id
    private Long userId;

    @Column
    private int point;

    @Builder
    public UserPointEntity(Long userId, int point) {
        this.userId = userId;
        this.point = point;
    }

    public UserPoint toModel() {
        return UserPoint.builder()
            .userId(userId)
            .point(point)
            .build();
    }
}

File: qc-core/qc-point/qc-point-rdb/src/main/kotlin/com/eager/questioncloud/point/entity/UserPointEntity.kt
package com.eager.questioncloud.point.entity

import com.eager.questioncloud.common.entity.BaseCustomIdEntity
import com.eager.questioncloud.point.domain.UserPoint
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "user_point")
class UserPointEntity(
    @Id var userId: Long,
    @Column var point: Int,
    isNewEntity: Boolean
) : BaseCustomIdEntity<Long>(isNewEntity) {
    fun toModel(): UserPoint {
        return UserPoint(userId, point)
    }
    
    companion object {
        fun createNewEntity(userPoint: UserPoint): UserPointEntity {
            return UserPointEntity(
                userPoint.userId,
                userPoint.point,
                true
            )
        }
    }
    
    override fun getId(): Long {
        return userId
    }
}
